### PR TITLE
Re-implement byId() endpoint

### PR DESCRIPTION
**Pull request by GTZL1**

### CODE REVIEW

File: src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java [hunk 1]
```diff
@@ -100,6 +96,45 @@ public QuestionController(QuestionRepository repository,
...
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") int idPoll,
+      @PathVariable(name = "idQuestion") int idQuestion)
+      throws ResourceNotFoundException, WrongCredentialsException {
+    ServerQuestionIdentifier idToFind = ServerQuestionIdentifier.builder()
+        .idServerQuestion(idQuestion)
+        .idxPoll(pollRepository.findById(ServerPollIdentifier.builder().idPoll(idPoll).idxModerator(
+            moderatorRepository.findById(idModerator).orElseThrow(ResourceNotFoundException::new))
+            .build())
+            .orElseThrow(ResourceNotFoundException::new))
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java:126]: Strange indentation, but the code looks correct to me 😄

File: src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java [hunk 2]
```diff
@@ -100,6 +96,45 @@ public QuestionController(QuestionRepository repository,
...
+        .build();
+
+    ServerQuestion questionWanted = repository.findById(idToFind)
+        .orElseThrow(ResourceNotFoundException::new);
+
+    for (ServerQuestion q : this.all(token, idModerator, idPoll)) {
+      if (q.equals(questionWanted)) {
+        return q;
+      }
+    }
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java]: I think it's probably better to check if the tokens match first, and then simply return the value found on lines 126 and 127.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java (hunk 1 above)
Comment (GTZL1) [reply src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java:126]: It validates checkstyle :)

File: src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java [hunk 3]
```diff
@@ -100,6 +97,71 @@ public QuestionController(QuestionRepository repository,
...
+    if (pollT.isEmpty()) {
+      //...it looks for a moderator
+      Optional<Moderator> pollM = modoRepo.findByToken(token);
+      if (pollM.isEmpty()) {
+        throw new ResourceNotFoundException();
+      } else {
+        //poll inside pollSet of the modo
+        pollTest = pollM.get().searchPoll(ServerPollIdentifier.builder()
+            .idPoll(idPoll).idxModerator(pollM.get()).build());
+      }
```
Comment (glsubri) [inline src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java]: Just so you know, you can directly have a poll like this:
```
Moderator pollM = modRepo.findByToken(token).orElseThrow(ResourceNotFoundException:new);
```

Comment (GTZL1) [reply src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java]: Yeah true

**Review by alexandrepiveteau — COMMENTED** [review]
Comments via Telegram DMs.

**Comment by alexandrepiveteau** [comment]
Included in #168.

**Pull request closed without merging**